_posts/2020-09-29-aws-developer-tools-mule-cicd-solution.md

```diff
@@ -0,0 +1,196 @@
+---
+layout: post
+title: Using AWS CodePipeline as CI/CD solution for MuleSoft Apps
+author: atteheino
+excerpt: Thoughts on using AWS Developer Tools to replace Jenkins as the CI/CD solution for MuleSoft Apps to be deployd to CloudHub
+tags:
+ - AWS CodeCommit
+ - AWS CodePipeline
+ - AWS CodeBuild
+ - MuleSoft
+ - Jenkins
+ - CloudHub
+ - CI/CD
+---
+
+A typical tooling setup for a integration project consists of a Git repository somewhere, Jenkins on some host, Application Server either on-premise or hosted somewhere. Git stores the source code, Jenkins builds and deploys the code to the application server running the application. 
+
+I was given the option to try out AWS alternative to this setup. AWS has [CodeCommit](https://aws.amazon.com/codecommit/) for source control, [CodeBuild](https://aws.amazon.com/codebuild/) for building your software and [CodePipeline](https://aws.amazon.com/codepipeline/) for automating the whole CI/CD process.
+
+Typically it would be a good idea to document your infrastructure as code, for example by creating the services you need via CloudFormation templates. Here I was doing a Proof Of Concept, so I did all of the setup via the AWS Console UI.
+
+# CodeCommit to store the source code
+
+It's not a requirement to use AWS CodeCommit, to be able to utilize CodeBuild or CodePipeline, but in my case it made sense as the original source code was not in a public repository like GitHub.
```

Comment:
Commas in this sentence are not necessary in my opinion, at least the one after CodeCommit.